[ruth-lim] iP

ruth-lim opened the pull request:
# Duke Pro

> "To do two things at once is to do neither." - Duke

Duke Chatbot is not just another task manager; it's your personal assistant to keep you organized and on top of your tasks. Here's why you'll love it:

- **Simplicity**: No muss, no fuss. Just straightforward task management. 
- **User-Friendly**: It's so easy to use that you'll be up and running in no time.
- **~~FAST~~ _LIGHTNING FAST_** ⚡⚡ : Duke is optimized for speed, making task management a breeze.

Getting started with Duke is a piece of cake:

1. [Download Duke](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. Double-click to launch.
3. Add your tasks.
4. Let Duke handle the rest! 😉
   
And the best part? It's absolutely **FREE**!


**Key Features**
- [X] Managing tasks
- [ ] Stay on top of your deadlines (*coming soon*)
- [ ] Set reminders to keep you on track (*coming soon*)


If you are using Java, here is the `main` method to run it:

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4 ToDos, Events, Deadlines

Commit:
A-TextUiTesting: Automated Text UI Testing

Commit:
Level-5: Handle Errors

Commit:
Level-5 Handle Errors

Commit:
Level-6 Delete

Commit:
A-Enums Use Enums + added more error handling

Commit:
Add Storage.java

feat: Implement saving of tasks in hard drive

* Implement loading and saving tasks from/to data file
* Handle corrupt or invalid data file

Commit:
Update to LocalDateTime for Deadline and Event

Commit:
Add CHECK and TODAY commands

Commit:
Add TaskList.java

Commit:
Add UI.java to handle interactions with the user

Commit:
Modify Storage.java to load and save tasks

Commit:
Add Parser.java

Commit:
Add *Command classes that inherit from an abstract Command class

Commit:
Divide classes into packages, all classes are under duke package

Commit:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552' into branch-A-Gradle

Commit:
Add JUnit tests for TaskList

Commit:
Add JUnit tests for Todo, Deadline and Event

Commit:
Add JUnit test for Storage

Commit:
Create JAR file

Commit:
Add JavaDoc comments for all classes.

Commit:
Add Find feature to find task in the list by keyword

Commit:
Update classes to comply with Java coding standard

Commit:
Merge branch 'branch-A-CodingStandard' into master

Commit:
Merge branch 'branch-Level-9'

Commit:
Extract task filtering logic from Ui into Command

Commit:
Add CheckStyle to detect coding style violations

Commit:
Correct coding style violations

Commit:
Setup JavaFX for project

Commit:
Update Duke to read and return String

Commit:
Style GUI

Commit:
Close platform and terminate program upon bye command

Commit:
Fix coding style violations

Commit:
Add Help command to display list of commands

Commit:
Fix variable naming and refactor CommandType class under command package

jacobcuison commented on src/main/java/duke/Parser.java:
```diff
@@ -34,13 +35,13 @@ public class Parser {
-        String[] parts = input.split(" ", 2);
-        String command = parts[0].toUpperCase();
+    public static Command parseUserInput(String input) throws DukeException {
+        String[] inputParts = input.split(" ", 2);
```
I am nitpicking, but perhaps this variable could be renamed. I was not very clear what it contained as I was reading through the switch statement below, but maybe that's just me!

jacobcuison commented on src/main/java/duke/Parser.java:
```diff
@@ -34,13 +35,13 @@ public class Parser {
-        String command = parts[0].toUpperCase();
+    public static Command parseUserInput(String input) throws DukeException {
+        String[] inputParts = input.split(" ", 2);
+        String userCommand = inputParts[0].toUpperCase();
```
Good job on naming your variables properly (i.e. singular and camel case)!

jacobcuison commented on src/main/java/duke/command/CommandType.java:
```diff
@@ -0,0 +1,19 @@
+    HELP,
+    LIST,
+    MARK,
+    TODAY,
```
I love how you thought of including other commands, such as `Today` and `Help`! Inspiring!

jacobcuison commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -6,33 +6,33 @@
 
     /**
-     * Constructs a TaskList with existing tasks.
+     * Constructs a task list with existing tasks.
```
Great job on following the format of JavaDocs!

jacobcuison commented on src/main/java/duke/Parser.java:
```diff
@@ -51,21 +52,21 @@ public static Command parseCommand(String input) throws DukeException {
             return new ListCommand();
         case MARK:
-            return new MarkCommand(Integer.parseInt(parseArgument(parts)) - 1);
+            return new MarkCommand(Integer.parseInt(parseArgument(inputParts)) - 1);
```
Perhaps you can include a comment here to explain the rationale of subtracting 1; on first read, the rationale may not be so obvious to someone who is unfamiliar with the app.

jacobcuison reviewed (commented):
Overall, I found your code well-formatted and very easy to read :) There were no obvious coding standard violations, just a few nits from my end. Great job Ruth! All the best!

Commit:
Improve code quality based on PR reviews

Commit:
Update build.gradle

Commit:
Add various asssertions

Commit:
Update FindCommand and Ui methods Using Streams

FindCommand and Ui methods uses traditional loops to iterate
through the list of tasks and manually build a message to display
to the user.

The code is less concise, more error-prone, and affects readability.

Using Streams abstracts away the low-level details of iteration and
string manipulation. It allows for a more declarative and functional
approach to processing the task list, resulting in cleaner and
more maintainable code.

Let's,
* Update FindCommand to use Streams for task list filtering.
* Update showList to use Streams for task list formatting.
* Update showTasksOnDate to use Streams for task list formatting.
* Update showTasksForToday to use Streams for task list formatting.
* Update showMatchingTask to use Streams for task list formatting.

Commit:
Fix coding quality issues

ruth-lim committed:
Merge pull request #1 from ruth-lim/branch-A-Assertions

Add various asssertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

ruth-lim committed:
Merge pull request #2 from ruth-lim/branch-A-CodeQuality

Fix coding quality issues

Commit:
Merge branch 'master' into branch-A-Streams

ruth-lim committed:
Merge pull request #3 from ruth-lim/branch-A-Streams

Update FindCommand and Ui methods using Streams

Commit:
Set up continuous integration

Commit:
Update code for collecting list from stream to follow Java 11

Commit:
Add Update Feature

Lacking in capability to modify tasks after they are created.
User has to delete an existing task if they want to change the
description or date of a task.

Introduce update feature to allow users to modify either descriptions
or date-times or both.
Specifically, for `Todo` tasks, it only allows for updates to
descriptions only.

Commit:
Fix checkstyle errors

Commit:
Fix JUnit test cases

ruth-lim committed:
Merge pull request #4 from ruth-lim/branch-C-Update

Add Update Feature

Commit:
Fix issue detected by bot

Commit:
Change duke and user profile picture

Commit:
Update MainWindow to be resizable

Commit:
Update UI colour

Commit:
Update FindCommand to be case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's,
* update the search algorithm to use case-insensitive matching

Commit:
Add representative screenshot of the chatbot to the `docs` folder

Commit:
Add Jekyll theme

ruth-lim committed:
Update _config.yml

ruth-lim committed:
Update README.md

ruth-lim committed:
Update _config.yml

ruth-lim committed:
Update _config.yml

Commit:
Remove Jekyll theme

ruth-lim committed:
Update README.md

Commit:
Update Ui messages

ruth-lim committed:
Fix README.md
